Updated the URL on how to create plugin and mention license. (#790)
freenas/freenas-docs: userguide/plugins.rst

Create a Plugin
---------------

Before creating a plugin, read the

> `LICENSE.md <https://github.com/ix-plugin-hub/documentation/blob/master/LICENSE.md>`__

Reviewer: Add period to end of line, but move the license section to after the introduction.

`iocage-plugin-quassel artifact repository
<https://github.com/freenas/iocage-plugin-quassel/blob/master/settings.json>`__.

> .. The above URL will need to be changed once Quasselcore is migated

Reviewer: ```suggestion
.. This URL will need to be changed when Quasselcore migrates
```

`iocage-plugin-quassel artifact repository
<https://github.com/freenas/iocage-plugin-quassel/blob/master/settings.json>`__.

> .. This URL will need to be changed when Quasselcore is migrated

Reviewer: This comment can be removed now.

Author: The plugin hasn't been migrated yet.

   `GitHub Guides <https://guides.github.com/>`__.

> On the local copy of :file:`ix-plugin-hub`, create a new file for

Reviewer: It's not clear what "a new file for the plugin to be added to %brand%" means.  Later context explains that this is a json file, but this sentence should be clarified.

   `GitHub Guides <https://guides.github.com/>`__.

On the local copy of :file:`iocage-ix-plugins`, create a new JSON file

> for the %brand% plugin with the fields described in

Reviewer: This does not explain what the file does. The naming convention should be mentioned next, because the user has to enter a name when they create the file.
`` A single JSON file describes the plugin, the packages it requires for operation, and other installation details. This file is named :samp:`{pluginname}.json`. For example, the Transmission(link) plugin file is named :file:`transmission.json`. ``

`INDEX <https://github.com/freenas/iocage-ix-plugins/blob/master/INDEX>`__
for examples of :file:`INDEX` entries.

> .. The INDEX URL above will need to be updated once the INDEX file is

Reviewer: Is this comment still needed?

Author: Yes. Hasn't been migrated yet.
